optuna_dashboard/ts/components/GraphTimeline.tsx

```diff
@@ -1,24 +1,27 @@
-import { Card, CardContent, Grid, Typography, useTheme } from "@mui/material"
-import * as Optuna from "@optuna/types"
+import { Grid, useTheme } from "@mui/material"
+import { PlotTimeline } from "@optuna/react"
 import * as plotly from "plotly.js-dist-min"
 import React, { FC, useEffect } from "react"
-import { StudyDetail, Trial } from "ts/types/optuna"
+import { StudyDetail } from "ts/types/optuna"
 import { PlotType } from "../apiClient"
-import { makeHovertext } from "../graphUtil"
+import { studyDetailToStudy } from "../graphUtil"
 import { usePlot } from "../hooks/usePlot"
 import { usePlotlyColorTheme } from "../state"
 import { useBackendRender } from "../state"
 
 const plotDomId = "graph-timeline"
-const maxBars = 100
 
 export const GraphTimeline: FC<{
   study: StudyDetail | null
 }> = ({ study }) => {
+  const theme = useTheme()
+  const colorTheme = usePlotlyColorTheme(theme.palette.mode)
+
   if (useBackendRender()) {
     return <GraphTimelineBackend study={study} />
   } else {
-    return <GraphTimelineFrontend study={study} />
+    const _study = studyDetailToStudy(study)
+    return <PlotTimeline study={_study} colorTheme={colorTheme} />
```

Comment:
A variable name beginning with `_` might give us the impression that it is unused, it may not be the most appropriate choice.

```suggestion
    return <PlotTimeline study={studyDetailToStudy(study)} colorTheme={colorTheme} />
```